src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -146,100 +199,224 @@
+        .define(FILENAME_OFFSET_ZERO_PAD_WIDTH_CONFIG, Type.INT, FILENAME_OFFSET_ZERO_PAD_WIDTH_DEFAULT, ConfigDef.Range.atLeast(0), Importance.LOW, FILENAME_OFFSET_ZERO_PAD_WIDTH_DOC,
+                CONNECTOR_GROUP, 11, Width.SHORT, FILENAME_OFFSET_ZERO_PAD_WIDTH_DISPLAY);
+
+    // Define Internal configuration group
+    config.define(STORAGE_CLASS_CONFIG, Type.STRING, STORAGE_CLASS_DEFAULT, Importance.LOW, STORAGE_CLASS_DOC, INTERNAL_GROUP, 1, Width.MEDIUM, STORAGE_CLASS_DISPLAY);
+  }
 
-  public static final String HDFS_NAMENODE_PRINCIPAL_CONFIG = "hdfs.namenode.principal";
-  private static final String HDFS_NAMENODE_PRINCIPAL_DOC = "The principal for HDFS Namenode.";
-  public static final String HDFS_NAMENODE_PRINCIPAL_DEFAULT = "";
+  private static class SchemaCompatibilityRecommender extends BooleanParentRecommender {
+    
+    public SchemaCompatibilityRecommender() {
+      super(HIVE_INTEGRATION_CONFIG);
+    }
+      
+    @Override
+    public List<Object> validValues(String name, Map<String, String> connectorConfigs) {
+      boolean hiveIntegration = Boolean.parseBoolean(connectorConfigs.get(parentConfigName));
+      if (hiveIntegration) {
+        return Arrays.<Object>asList("BACKWARD", "FORWARD", "FULL");
+      } else {
+        return Arrays.<Object>asList("NONE", "BACKWARD", "FORWARD", "FULL");
+      }
+    }
+
+    @Override
+    public boolean visible(String name, Map<String, String> connectorConfigs) {
+      return true;
+    }
+  }
+  
+  private static class BooleanParentRecommender implements ConfigDef.Recommender {
+    
+    protected String parentConfigName;
+    
+    public BooleanParentRecommender(String parentConfigName) {
+      this.parentConfigName = parentConfigName;
+    }
+    
+    @Override
+    public List<Object> validValues(String name, Map<String, String> connectorConfigs) {
+      return new LinkedList<>();
+    }
+
+    @Override
+    public boolean visible(String name, Map<String, String> connectorConfigs) {
+      return Boolean.parseBoolean(connectorConfigs.get(parentConfigName));
+    }
+  }
 
-  public static final String KERBEROS_TICKET_RENEW_PERIOD_MS_CONFIG = "kerberos.ticket.renew.period.ms";
-  private static final String KERBEROS_TICKET_RENEW_PERIOD_MS_DOC =
-      "The period in milliseconds to renew the Kerberos ticket.";
-  public static final long KERBEROS_TICKET_RENEW_PERIOD_MS_DEFAULT = 60000 * 60;
+  private static class PartitionerClassDependentsRecommender implements ConfigDef.Recommender {
+
+    @Override
+    public List<Object> validValues(String name, Map<String, String> props) {
+      return new LinkedList<>();
+    }
+
+    @Override
+    public boolean visible(String name, Map<String, String> props) {
+      HdfsSinkConnectorConfig connectorConfig = new HdfsSinkConnectorConfig(props);
```

Comment:
Are we guaranteed this will parse correctly by the time any calls to `visible` are made? This seems a bit wasteful since we end up needing to parse the entire config to check visibility for each config. Maybe this isn't a problem in practice? Or should the parameter actually be different? I think this might come back to the question of whether an `AbstractConfig` subclass should be used instead of `ConfigDef`.